include/istio/control/http/check_data.h

```diff
@@ -47,6 +47,9 @@ class CheckData {
   // Returns true if connection is mutual TLS enabled.
   virtual bool IsMutualTLS() const = 0;
 
+  // Get requested server name, SNI in case of TLS
+  virtual std::string GetRequestedServerName() const = 0;
```

Comment:
Since there are cases data is not available,   the coding style is to return a bool.  It should be:

// return true if data is valid.
bool GetRequestedServerName(std::string* name);